### PR TITLE
BUG: rename fails when using level

### LINKED ISSUE #43659: BUG: rename fails when using level
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

arrays = [
        ["bar", "bar", "baz", "baz", "foo", "foo", "qux", "qux"],
        ["one", "two", "one", "two", "one", "two", "one", "two"],
]
tuples = list(zip(*arrays))
index = pd.MultiIndex.from_tuples(tuples, names=["first", "second"])
s = pd.Series(np.random.randn(8), index=index)
s.rename(index={'one': 'yes'}, level='second', errors='raise')
```

### Issue Description

`rename` does never find index values when using the `level` parameter to replace. When adding `errors='raise'`, we get the following error:
```
Traceback (most recent call last):
  File "/Users/bustawin/repos/venvs/bicing/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-15-54b8de4a3511>", line 1, in <module>
    s.rename(index={'one': 'yes'}, level=1, errors='raise')
  File "/Users/bustawin/repos/venvs/bicing/lib/python3.9/site-packages/pandas/core/series.py", line 4518, in rename
    return super().rename(
  File "/Users/bustawin/repos/venvs/bicing/lib/python3.9/site-packages/pandas/core/generic.py", line 1154, in rename
    raise KeyError(f"{missing_labels} not found in axis")
KeyError: "['one'] not found in axis"
```

The example dataframe is taken from the [multi-index guide](https://pandas.pydata.org/docs/user_guide/advanced.html#creating-a-multiindex-hierarchical-index-object).

### Expected Behavior

The above should have produced the following dataframe:
```
first  second
bar    yes       0.220890
       two      -1.497161
baz    yes       0.589343
       two      -0.604643
foo    yes       0.090448
       two       0.373825
qux    yes       1.366876
       two       1.460868
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 73c68257545b5f8530b7044f56647bd2db92e2ba
python           : 3.9.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.3.3
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.1.3
setuptools       : 57.0.0
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : None
psycopg2         : 2.9.1 (dt dec pq3 ext lo64)
jinja2           : 3.0.1
IPython          : 7.27.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 2021.08.1
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.7.1
sqlalchemy       : 1.3.24
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.53.1

</details>

### PR DESCRIPTION
**Pull request by wojtek2kdev**
- [x] closes #43659 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/generic/test_series.py [hunk 1]
```diff
@@ -151,3 +151,28 @@ def finalize(self, other, method=None, **kwargs):
         # reset
         Series._metadata = _metadata
         Series.__finalize__ = _finalize  # FIXME: use monkeypatch
+
+    # issue #43659
+    def test_rename_with_multiindex(self):
+        arrays = [
```
Comment (phofl) [inline pandas/tests/generic/test_series.py:157]: Please add reference in test not above

The test is to complicated. Please reduce number of rows and don't use something like list(zip(*arrays)). You can use MultiIndex.from_arrays

**Comment by phofl** [comment]
Also need whatsnew

**Comment by wojtek2kdev** [comment]
@phofl Sorry, it's my first PR to this project. What is "whatsnew entry"?

**Comment by phofl** [comment]
Check doc/source/whatsnew. These are our release notes. Yours should go into 1.4

**Comment by wojtek2kdev** [comment]
@phofl I'm not sure, but all of failed checks are my fault? For example: https://github.com/pandas-dev/pandas/pull/43667/checks?check_run_id=3655863372. I checked this test locally and it passes.

File: pandas/tests/generic/test_series.py [hunk 2]
```diff
@@ -151,3 +151,26 @@ def finalize(self, other, method=None, **kwargs):
         # reset
         Series._metadata = _metadata
         Series.__finalize__ = _finalize  # FIXME: use monkeypatch
+
+    def test_rename_with_multiindex(self):
```
Comment (jreback) [inline pandas/tests/generic/test_series.py:155]: move here: pandas/tests/series/methods/test_rename.py

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_rename.py [hunk 3]
```diff
@@ -101,3 +102,26 @@ def test_rename_callable(self):
...
+            ["bar", "baz", "baz", "foo", "qux"],
+            ["yes", "yes", "two", "two", "yes"],
+        ]
+
+        index_expected = MultiIndex.from_arrays(
+            arrays_expected, names=["first", "second"]
+        )
+        series_expected = Series(np.ones(5), index=index_expected)
+
+        assert result.equals(series_expected)
```
Comment (jreback) [inline pandas/tests/series/methods/test_rename.py:127]: can you use `tm.assert_series_equal(result, expected)`

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_rename.py (hunk 3 above)
Comment (wojtek2kdev) [reply pandas/tests/series/methods/test_rename.py]: Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @wojtek2kdev very nice!

**Pull request merged**